bblocks/import_tools/world_bank.py

```diff
@@ -231,98 +215,151 @@ def _clean_pink_sheet(df: pd.DataFrame, sheet: str) -> pd.DataFrame:
+            self.update_data and self.data is None
+        ):
             self.update()
-        else:
-            self.data = pd.read_csv(f"{PATHS.imported_data}/{self.file_name}")
 
-    @property
-    def file_name(self) -> str:
-        """Returns the name of the stored file"""
-        return f"World Bank Pink Sheet - {self.sheet}.csv"
+        self.__load(indicator)
+
+        return self
+
+    def update(self, reload_data=False) -> ImportData:
+        """Update the data saved on disk
+
+        When called it downloads Pink sheet Data from the World Bank and saves it to disk.
+        Optionally specify whether to reload the data to the object
+
+        Args:
+            reload_data: If True, the data will be reloaded to the object
+
+        Returns:
+            The same object to allow chaining
+        """
+
+        url = (
+            "https://thedocs.worldbank.org/en/doc/5d903e848db1d1b83e0ec8f744e55570-0350012021/related/CMO"
+            "-Historical-Data-Monthly.xlsx"
+        )
 
-    def update(self) -> None:
-        """Updates the underlying data"""
         (
-            _read_pink_sheet(self.sheet)
-            .pipe(_clean_pink_sheet, self.sheet)
+            pd.read_excel(url, sheet_name=self.sheet)
+            .pipe(_clean_pink_sheet, sheet=self.sheet)
             .to_csv(f"{PATHS.imported_data}/{self.file_name}", index=False)
         )
 
-        self.data = pd.read_csv(f"{PATHS.imported_data}/{self.file_name}")
+        if reload_data:
+            self.__load(indicator_list=list(self.indicators))
+
+        return self
 
     def get_data(
         self,
-        *,
+        indicators: str | list = None,
         start_date: str = None,
         end_date: str = None,
-        indicators: str | list = None,
-    ):
-        """Get data as a Pandas DataFrame
+    ) -> pd.DataFrame:
+        """Get the data as a Pandas DataFrame
 
         Args:
-            start_date : start date of the data to be returned (inclusive).
-            end_date : end date of the data to be returned (inclusive).
-            indicators : one or more indicators to be returned.
-
+            indicators: indicator to get. If 'all', both indicators will be returned
+            start_date: start date of the data (e.g. '2019-01-01')
+            end_date: end date of the data (e.g. '2019-12-31')
 
         Returns:
-            pd.DataFrame: Pandas dataframe with the requested data
+            Pandas DataFrame of the data
         """
 
-        self.data = pd.read_csv(f"{PATHS.imported_data}/{self.file_name}")
+        df = self.data.copy(deep=True)
+
+        if indicators is not None:
+            if isinstance(indicators, str):
+                indicators = [indicators]
+            for indicator in indicators:
+                if indicator not in df.indicator.unique():
+                    raise Warning(f"indicator not found: {indicator}")
+            df = df[df.indicator.isin(indicators)]
 
```

Comment:
This implementation doesn't make use of the fact that the indicators have been loaded to the `indicators` variable.

`self.data` should always hold the raw, full CSV.

When the user asks for the data, the object should:
- check that the indicators listed in `indicators` exist as keys inside the `self.indicators` variable (if `indicators` isn't 'all').
- Return a dataframe which appends the requested data (vertically).

Right now, the object is quite simple and deals only with monthly data. But a future version could add more functionality (like transforming to yearly or quarterly averages, for example). That's why your getter shouldn't duplicate tasks that have already been performed on `load_indicator`. It is also the main reason to separate object creation from data loading, and from data getting.

`start_date` and `end_date` appear to be optional parameters. They should be labelled as such. If I follow correctly, if they are not passed, you get the full dataset. If only one is passed, you apply it to the start or end of the series, and if both, you capture the data between the dates.


Please modify the getter accordingly